packages/components/src/navigator/navigator-provider/component.tsx

```diff
@@ -25,18 +27,34 @@ function NavigatorProvider(
 	props: WordPressComponentProps< NavigatorProviderProps, 'div' >,
 	forwardedRef: Ref< any >
 ) {
-	const { initialPath, children, ...otherProps } = useContextSystem(
-		props,
-		'NavigatorProvider'
-	);
+	const {
+		initialPath,
+		children,
+		className,
+		...otherProps
+	} = useContextSystem( props, 'NavigatorProvider' );
 
-	const [ path, setPath ] = useState< NavigatorPath >( {
+	const [ navigatorPath, setNavigatorPath ] = useState< NavigatorPath >( {
 		path: initialPath,
 	} );
+	const [ isAnimating, setIsAnimating ] = useState< boolean >( false );
+
+	const cx = useCx();
+	const classes = useMemo(
+		() => cx( isAnimating && css( { overflow: 'hidden' } ), className ),
+		[ className, isAnimating ]
+	);
 
 	return (
-		<View ref={ forwardedRef } { ...otherProps }>
-			<NavigatorContext.Provider value={ [ path, setPath ] }>
+		<View ref={ forwardedRef } className={ classes } { ...otherProps }>
+			<NavigatorContext.Provider
+				value={ {
+					navigatorPath,
+					setNavigatorPath,
+					isAnimating,
+					setIsAnimating,
```

Comment:
Naming opinion: `navigatorPath` and `setNavigatorPath` seem unnecessarily specific/verbose. I think `location` and `navigate` would go well.

Reply:
That is a fair point. I took your advice and tweaked it a little in [`814b03d` (#35317)](https://github.com/WordPress/gutenberg/pull/35317/commits/814b03d6d661812842fe578d92826e2f88849713)

`navigatorPath` => `location`
`setNavigatorPath` => `setLocation`

I think `setLocation` is more clear than `navigate` about its relationship with `location`